assets/control-plane/service-monitor-kubelet-prometheus-adapter.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app.kubernetes.io/name: kubelet
+    app.kubernetes.io/part-of: openshift-monitoring
+    k8s-app: kubelet
+  name: kubelet-for-pa
+  namespace: openshift-monitoring
+spec:
+  endpoints:
+  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    honorLabels: true
+    honorTimestamps: true
+    interval: 30s
+    metricRelabelings:
+    - action: drop
+      regex: container_(network_tcp_usage_total|network_udp_usage_total|tasks_state|cpu_load_average_10s)
+      sourceLabels:
+      - __name__
+    - action: drop
+      regex: (container_spec_.*|container_file_descriptors|container_sockets|container_threads_max|container_threads|container_start_time_seconds|container_last_seen);;
+      sourceLabels:
+      - __name__
+      - pod
+      - namespace
+    - action: drop
+      regex: (container_blkio_device_usage_total);.+
+      sourceLabels:
+      - __name__
+      - container
+    - action: keep
+      regex: container_cpu_usage_seconds_total|container_memory_working_set_bytes
+      sourceLabels:
+      - __name__
+    - action: replace
+      replacement: pa_$1
+      sourceLabels:
+      - __name__
+      targetLabel: __name__
+    path: /metrics/cadvisor
```

Comment:
Targeting `/metrics/resources` should be more efficient on Prometheus side because we won't have to drop any metrics. It might also be less demanding on the kubelet side.

https://kubernetes.io/docs/concepts/cluster-administration/system-metrics/

Reply:
the other advantage is that we can scope the prometheus-adapter's queries using the `metrics_path` label (e.g. `container_cpu_usage_seconds_total{metrics_path="/metrics/resources",...}`)

Reply:
From what I understand reading the kubelet code base, the `/metrics/resource` handler should be more lightweight than cAdvisor in terms of resource usage.

Reply:
yes there's a risk but we should be able to find which recording rules need to be fixed. At the same time, having a different metric name is indeed more obvious.

Reply:
I added the `/metrics/resource` endpoint but kept the metrics prefix. wdyt?